docs/2-fork-accountability.md

```diff
@@ -0,0 +1,191 @@
+
+The critical case outlined above is when a process sends a PRECOMMIT message for a value *v* and then, in a later round, it sends a PREVOTE message for a different value than *v*. In this way, a faulty process can lead to a fork without making equivocation.
+
+An asynchronous accountability algorithm would work correctly if processes can justify the sending of a PREVOTE message directly: a simple solution would be attaching the justifications to the PREVOTE message itself so that the accountability algorithm can directly check if the message has been correctly sent.
+In this case, the monitor would be able to complete the accountability algorithm correctly (respecting both the completeness and accuracy properties) as soon as at least *f + 1* faulty processes will be detected and by relying on the fact that at least *f + 1* correct processes will send correct message logs.
+
+## Algorithm design
+ 
+Assuming the small modification in the Tendermint consensus algorothm described above, these are the high-level steps carried out by the monitor to run the asynchronous accountability algorithm:
+
+1. Upon a fork, monitor requests message logs from all the validators
+
+2. Monitor runs the accountability algorithm to detect faulty processes upon receiving at least *f + 1* message logs and at every new message logs received when there are at least *f + 1* message logs 
+
+3. While running the accountability algorithm, the monitor analyzes the received logs: for every message *m* that has been received by some other process but is not present in the sender's message log, the monitor attaches *m* to the message log of the sender of *m*
+
+4. For each validator, the monitor scans the received message logs and the inferred message logs and determines for each process whether it is faulty or not by analyzing the history of sent messages and transitions
+
+5. When the monitor detects at least *f + 1* faulty processes, the algorithm completes
+
+Termination is guaranteed by the fact that correct processes will send their messages logs and their message logs will be correct (no sent message will be missing). 
+If this condition does not hold, the algorithm would not be able to identify correctly all the faulty processes that led to a fork. However, correct processes have no reason to misbehave or send invalid message logs. 
+
+## Notes
+
+- As some readers might have noticed, the fork accountability problem can be easily solved by simply piggybacking justifications in every single message. 
+However, this solution would be inefficient because it considerably increases the amount of information stored and exchanged by processes. The scope of the discussion is to solve the problem efficiently with a minimal amount of changes required on the Tendermint consensus protocol and that is the reason why we do not consider and analyze this possible solution.   
```

Comment:
We should discuss this today.

Reply:
Added "future work" section to better explain this.